network/p2p/p2p.go

```diff
@@ -172,6 +201,43 @@ func (n *p2pNetwork) Close() error {
 	return n.host.Close()
 }
 
+func (n *p2pNetwork) getConnector(logger *zap.Logger) (chan peer.AddrInfo, error) {
+	connector := make(chan peer.AddrInfo, connectorQueueSize)
+	go func() {
+		// Wait for own subnets to be subscribed to and updated.
+		// TODO: wait more intelligently with a channel.
+		time.Sleep(8 * time.Second)
+		ctx, cancel := context.WithCancel(n.ctx)
+		defer cancel()
+		n.backoffConnector.Connect(ctx, connector)
+	}()
+
+	// Connect to trusted peers first.
+	go func() {
+		for _, addrInfo := range n.trustedPeers {
+			connector <- *addrInfo
+		}
+	}()
+
+	ma, err := peer.AddrInfoToP2pAddrs(&peer.AddrInfo{
+		ID:    n.host.ID(),
+		Addrs: n.host.Addrs(),
+	})
+	if err != nil {
+		logger.Fatal("could not get my address", zap.Error(err))
+		return nil, err
+	}
+	maStrs := make([]string, len(ma))
+	for i, ima := range ma {
+		maStrs[i] = ima.String()
+	}
+	logger.Info("starting p2p",
+		zap.String("my_address", strings.Join(maStrs, ",")),
+		zap.Int("trusted_peers", len(n.trustedPeers)),
+	)
```

Comment:
this part is unrelated to getConnector, could be extracted to logStartup or just moved back into Start()